proving_system/stark/src/verifier.rs

```diff
@@ -178,111 +247,86 @@ where
-    }
-
-    // // END TRACE <-> Composition poly consistency evaluation check
-
-    let lde_root_order =
-        (air.context().trace_length * air.options().blowup_factor as usize).trailing_zeros();
-
-    // Get the number of trace terms the DEEP composition poly will have.
-    // One coefficient will be sampled for each of them.
-    // TODO: try remove this, call transcript inside for and move gamma declarations
-    let trace_term_coeffs = &(0..trace_poly_ood_evaluations.num_columns())
-        .map(|_| {
-            (0..trace_poly_ood_evaluations.num_rows())
-                .map(|_| transcript_to_field(transcript))
-                .collect()
-        })
-        .collect::<Vec<Vec<FieldElement<F>>>>();
-
-    // Get coefficients for even and odd terms of the composition polynomial H(x)
-    let gamma_even = &transcript_to_field::<F, _>(transcript);
-    let gamma_odd = &transcript_to_field::<F, _>(transcript);
-
-    // construct vector of betas
-    let mut beta_list = Vec::new();
-    let count_betas = proof.fri_layers_merkle_roots.len() - 1;
-
-    for (i, merkle_roots) in proof.fri_layers_merkle_roots.iter().enumerate() {
-        let root = merkle_roots.clone();
-        let root_bytes = root.to_bytes_be();
-        transcript.append(&root_bytes);
-
-        if i < count_betas {
-            let beta = transcript_to_field(transcript);
-            beta_list.push(beta);
-        }
-    }
-
-    // DEEP consistency check
-    // 1. Verify that Deep(x) is constructed correctly
-    let last_evaluation = &proof.query_list[0].fri_decommitment.last_layer_evaluation;
-    let last_evaluation_bytes = last_evaluation.to_bytes_be();
-    transcript.append(&last_evaluation_bytes);
-
-    let q_0 = transcript_to_usize(transcript) % (2_usize.pow(lde_root_order));
-    transcript.append(&q_0.to_be_bytes());
-
-    let deep_composition_poly_args = &mut DeepCompositionPolyArgs {
-        root_order,
-        trace_term_coeffs,
-        gamma_even,
-        gamma_odd,
-        d_evaluation_point: &lde_roots_of_unity_coset[q_0],
-        ood_evaluation_point: &z,
-        trace_poly_ood_evaluations,
-        composition_poly_ood_evaluations,
-        deep_consistency_check: &proof.deep_consistency_check,
-    };
-
-    let deep_poly_evaluation = compare_deep_composition_poly(deep_composition_poly_args);
-    let deep_poly_claimed_evaluation = &proof.query_list[0].fri_decommitment.layer_evaluations[0].0;
-
-    if deep_poly_claimed_evaluation != &deep_poly_evaluation {
-        return false;
-    }
+    composition_poly_claimed_ood_evaluation == composition_poly_ood_evaluation
+}
 
+fn step_3_verify_fri<F, T, A>(
+    air: &A,
+    proof: &StarkProof<F>,
+    domain: &Domain<F>,
+    challenges: &Challenges<F>,
+    transcript: &mut T,
+) -> bool
+where
+    F: IsTwoAdicField,
+    FieldElement<F>: ByteConversion,
+    T: Transcript,
+    A: AIR<Field = F>,
+{
     // 2. Verify that t(x_0) is a trace evaluation
     // 3. Verify first layer of FRI
```

Comment:
Because of the refactor, these steps were out of order. We can remove the numbers.